diff --git a/node_modules/mongoose-to-swagger/dist/index.js b/node_modules/mongoose-to-swagger/dist/index.js
index 52d1ea1..32abdd6 100644
--- a/node_modules/mongoose-to-swagger/dist/index.js
+++ b/node_modules/mongoose-to-swagger/dist/index.js
@@ -14,7 +14,7 @@ const mapMongooseTypeToSwaggerType = (type) => {
     if (type.schemaName === 'Mixed') {
         return 'object';
     }
-    if (type === 'ObjectId' || type === 'ObjectID') {
+    if (type === 'ObjectId' || type === 'ObjectID' || type === 'schemaobjectid') {
         return 'string';
     }
     if (type === bson_1.ObjectId) {
@@ -27,7 +27,7 @@ const mapMongooseTypeToSwaggerType = (type) => {
         return 'map';
     }
     if (type instanceof Function) {
-        if (type.name === 'ObjectId' || type.name === 'ObjectID') {
+        if (type.name === 'ObjectId' || type.name === 'ObjectID' || type.name === "SchemaObjectId") {
             return 'string';
         }
         if (type.name === 'Date') {
@@ -36,6 +36,9 @@ const mapMongooseTypeToSwaggerType = (type) => {
         if (type.name === 'Decimal128') {
             return 'number';
         }
+        if (type.name === 'Buffer') {
+            return 'string';
+        }
         return type.name.toLowerCase();
     }
     if (type.type != null) {
@@ -55,6 +58,7 @@ const mapMongooseTypeToSwaggerType = (type) => {
             case 'String':
             case 'SchemaString':
             case 'SchemaBuffer':
+            case 'schemaobjectid':
             case 'SchemaObjectId':
                 return 'string';
             case 'SchemaArray':
